### ISSUE TITLE
Error highlight bugs in Editor

### ISSUE THREAD

**Issue by mougino** [RFO-BASIC/Basic#183]
The following BASIC! codes all throw an Extraneous character error, but when back to the Editor, the incriminated line is not highlighted:

1) No highlight at all:
```Basic
REM Start of BASIC! Program
print a+
```

2) Highlight on the wrong line (the line before)
```Basic
REM Start of BASIC! Program
print a
print a+
```

I thought that the problem comes from the incriminated line not finishing with a Line Feed (if you add an empty line after the last line in these 2 programs, the bad line is correctly highlighted) but this last 3-line example shows it's not always working:

3) No highlight at all:
```Basic

print a+

```

**Comment by jMarcS** [comment]
There are two different errors here.

One is in `Basic.java:loadProgramFromString`. If there is no newline on the last line, `AddProgramLine.charCount` is not updated.

The other is in `Editor.java:onResume`, where the start and end are not always calculated correctly. Nicolas, if you are reading here, this may be another bug caused by your favorite Java quirk: if you have `start` and `end` parameters, `start` is the starting index, and `end` is the ending index __+1__.

I changed a lot of spaces to tabs in `Editor.java`, so use -w on diff.

**Comment by mougino** [comment]
I read you, thanks for the explanation, and good debug job as always Marc ;)

When you say to use -w on diff, is it the git command or the linux command ?
Because, you know... I use Windows ;)

**Comment by jMarcS** [comment]
Well, shame on you!

No, seriously, I am in my office so I am on Windows, too.

When you install git on Windows, you have the option of installing git bash. That's where you do git commands if you're not using the GUI. The git commands often look a lot like Linux commands. So, for example, git diff -w ignores whitespace.

So, for example, compare the output of these two lines:

    git diff 4ef202e 5055c42 -- src/com/rfo/basic/Editor.java
    git diff -w 4ef202e 5055c42 -- src/com/rfo/basic/Editor.java

I got the commit hashes from GitHub, but you can get them from gitk or wherever you like.

You can probably do all that with the GUI, too, but I don't know how :{)}

**Comment by jMarcS** [comment]
Released in v01.86, 2015/04/11.
